pkg/remoteimage/pull.go

```diff
@@ -40,14 +43,31 @@ func (p puller) ImageSize(ctx context.Context) int {
 }
 
 func (p puller) Pull(ctx context.Context) (err error) {
+	startTime := time.Now()
```

Comment:
this is the only real change to the actual pull code... how metrics are emitted in a centralized, consistent way regardless of async/sync